Spigot-Server-Patches/0460-Brand-support.patch

```diff
@@ -0,0 +1,73 @@
+From e6486d0432d03797a6deaad2c6fbd5de7ef8a0d3 Mon Sep 17 00:00:00 2001
+From: DigitalRegent <misterwener@gmail.com>
+Date: Mon, 6 Apr 2020 20:29:55 +0200
+Subject: [PATCH] Brand support
+
+
+diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
+index 7929fcc8..a559768e 100644
+--- a/src/main/java/net/minecraft/server/PlayerConnection.java
++++ b/src/main/java/net/minecraft/server/PlayerConnection.java
+@@ -106,6 +106,8 @@ public class PlayerConnection implements PacketListenerPlayIn {
+     private static final int MAX_SIGN_LINE_LENGTH = Integer.getInteger("Paper.maxSignLength", 80);
+     private static final long KEEPALIVE_LIMIT = Long.getLong("paper.playerconnection.keepalive", 30) * 1000; // Paper - provide property to set keepalive limit
+ 
++    private String clientBrandName = null; // Paper - Brand name
++
+     public PlayerConnection(MinecraftServer minecraftserver, NetworkManager networkmanager, EntityPlayer entityplayer) {
+         this.minecraftServer = minecraftserver;
+         this.networkManager = networkmanager;
+@@ -2641,6 +2643,10 @@ public class PlayerConnection implements PacketListenerPlayIn {
+     private static final MinecraftKey CUSTOM_REGISTER = new MinecraftKey("register");
+     private static final MinecraftKey CUSTOM_UNREGISTER = new MinecraftKey("unregister");
+ 
++    // Paper start
++    private static final MinecraftKey MINECRAFT_BRAND = new MinecraftKey("brand");
++    // Paper end
++
+     @Override
+     public void a(PacketPlayInCustomPayload packetplayincustompayload) {
+         PlayerConnectionUtils.ensureMainThread(packetplayincustompayload, this, this.player.getWorldServer());
+@@ -2664,6 +2670,10 @@ public class PlayerConnection implements PacketListenerPlayIn {
+                 PlayerConnection.LOGGER.error("Couldn\'t unregister custom payload", ex);
+                 this.disconnect("Invalid payload UNREGISTER!");
+             }
++        // Paper start - add brand support
```

Comment:
You should call the `else ` branch, too. You might be breaking some plugins that listens to this channel.

Reply:
thank you! fixed.